### PR TITLE
Feature/create artemis test module

### LINKED ISSUE #3833: ArtemisTestResource available
**Description**
Artemis integration test has an `ArtemisTestResource` https://github.com/quarkusio/quarkus/pull/2930/files#diff-76fa3fe6358cacc251f9cb4d29ae82ac to manage the lifecycle of an embedded Artemis instance for testing purposes.

The problem is that this class is not available for all users, I think that this kind of utilities should be available for any user because the way we test these integrations might be used by any dev.

So I think that this class should be moved to either an `artemis-test` artifact or in a place that can be used by anyone.

Generally speaking, I think that any util used in our integration tests should not be implemented in the integration tests itself but in a place that could be used by anyone, because at the end our necessities for testing probably are going to be the same for anyone writing test for Quarkus.

### PR DESCRIPTION
**Pull request by uaihebert**
Fixes #3833
The idea of this pr is:
1) create the test module with ArtemisResourceTest
1) expose it to be used by the developers that use Quarkus
1) replace integration tests that used the test class

### CODE REVIEW

File: extensions/artemis-test/pom.xml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-build-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../../build-parent/pom.xml</relativePath>
+    </parent>
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>quarkus-artemis-test</artifactId>
```
Comment (aguibert) [inline extensions/artemis-test/pom.xml]: we should put this test module under `quarkus/test-framework/` with the other test modules, and set the pom parent to `quarkus-test-framework`

**Review by aguibert — COMMENTED** [review]
Overall looks good, my only comment is on placement of the new module

**Review by gastaldi — CHANGES_REQUESTED** [review]
I agree with @aguibert's comment in https://github.com/quarkusio/quarkus/pull/5900/files#r352410389. The module should be moved under `quarkus-test-framework`

**Comment by gsmet** [comment]
Agreed and you will need to name it `quarkus-test-artemis` to be consistent with the others.

**Comment by uaihebert** [comment]
Thank you for the feedback @aguibert @gastaldi @gsmet 

I updated it as requested.

**Comment by gastaldi** [comment]
That looks good. Can you squash all your commits in a single one? Thank you!

**Comment by uaihebert** [comment]
@gastaldi sure, no problem.

**Review by gastaldi — APPROVED** [review]

**Comment by uaihebert** [comment]
hello @gastaldi the build failed on the windows test.

any tip for what should I look in the log? the logs are huge and I could not find the error in there.

**Comment by gastaldi** [comment]
@uaihebert it doesn't seem related to your changes, let me see if I can trigger the build again

**Comment by uaihebert** [comment]
hello @gastaldi indeed it was not related.

let me know if you need anything else.

**Comment by gastaldi** [comment]
@uaihebert Merged. Thank you!

**Comment by uaihebert** [comment]
thank you too.

**Pull request merged**